src/mbgl/util/geo.cpp

```diff
@@ -28,4 +28,16 @@ LatLngBounds::LatLngBounds(const TileID& id)
       ne(TileID{ id.z, id.x + 1, id.y, id.sourceZ }) {
 }
 
+PrecisionPoint EdgeInsets::getCenter(uint16_t width, uint16_t height) const {
+    return {
+        (width - left - right) / 2.0f + left,
+        (height - top - bottom) / 2.0f + top,
+    };
+}
+
+void EdgeInsets::flip() {
+    std::swap(top, bottom);
+    std::swap(left, right);
```

Comment:
[Some discussion on this line](https://github.com/mapbox/mapbox-gl-native/commit/eaeef958a5357713dd19c7031789399d6a883a76#commitcomment-15509284).